src/core/createCookieProxy.js

```diff
@@ -52,7 +55,7 @@ export default (name, expires, domain = "") => {
       deserializedCookie = updatedCookie;
       cookie.set(name, updatedCookie, {
         expires,
-        domain: domain || getTopLevelCookieDomain(window, cookie)
+        domain: domain || memoizedGetTopLevelCookieDomain(window, cookie)
```

Comment:
This is a simple enough use case that memoize probably isn't necessary, but I thought it would be nice anyway and I imagine we'll end up using the util elsewhere.